pkg/segment/writer/metrics/metricssegment.go

```diff
@@ -522,6 +527,38 @@ func ExtractOTSDBPayload(rawJson []byte, tags *TagsHolder) ([]byte, float64, uin
 						ts = uint32(intVal)
 					}
 				}
+			case jp.String:
+				// First, try to parse the date as a number (seconds or milliseconds since epoch)
+				if t, err := strconv.ParseInt(string(value), 10, 64); err == nil {
+					// Determine if the number is in seconds or milliseconds
+					if toputils.IsTimeInMilli(uint64(t)) {
+						ts = uint32(t / 1000)
+					} else {
+						ts = uint32(t)
+					}
+				}
+
+				// Parse the string to time using time.Parse and multiple layouts.
+				layouts := []string{
+					time.RFC3339,
+					time.RFC3339Nano,
+					time.RFC1123,
+					time.RFC1123Z,
+					time.RFC822,
+					time.RFC822Z,
+					time.RFC850,
```

Comment:
you are defining this layouts arr again and again during parsing, it will put pressure on GC. you should make this a const at the top of this file and just reuse it